### PR TITLE
Integrate auth API and refresh capabilities 

### PR DESCRIPTION
**Pull request by ruks**
- Support auth api
- Support SAS token renew

### CODE REVIEW

File: component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/auth/AuthClient.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (tishan89) [inline component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/auth/AuthClient.java:2]: Should be 2021

File: component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/client/EventHubProducerClientFactory.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    private static final Logger log = Logger.getLogger(EventHubClient.class);
+
+    public static EventHubProducerClient create(String authEndpoint, String authToken) {
+        TokenCredential tokenCredential = new WSO2TokenCredential(authEndpoint, authToken);
+        String tempSASToken;
+        try {
+            // generate SAS token to get eventhub meta data
+            tempSASToken = getSASToken(authEndpoint, authToken);
+        } catch (AuthenticationException e) {
+            throw new RuntimeException("SAS token generation failed.");
```
Comment (tishan89) [inline component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/client/EventHubProducerClientFactory.java:44]: lets return null here and in the EventhubClient use that as a flag in the sendEvent()

File: component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/client/WSO2TokenCredential.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    @Override
+    public Mono<AccessToken> getToken(TokenRequestContext tokenRequestContext) {
+        log.debug("Trying to retrieving a new SAS token.");
+        try {
+            String sasToken = AuthClient.getSASToken(this.authEndpoint, this.authToken);
+            log.debug("New SAS token retrieved.");
+            // Using lower duration than actual.
+            OffsetDateTime time = OffsetDateTime.now(ZoneOffset.UTC).plus(Duration.ofHours(20));
```
Comment (tishan89) [inline component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/client/WSO2TokenCredential.java:53]: We should parameterize this based on the token validity. We can move it to next milestone though. Please create an issue to track

**Review by tishan89 — CHANGES_REQUESTED** [review]

File: component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/client/WSO2TokenCredential.java (hunk 3 above)
Comment (ruks) [reply component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/client/WSO2TokenCredential.java:53]: ahh yes. I thought to improve it in the Auth API to send validity period or use the expiry time in the SAS token. Will discuss the best approach. 
Added a issue https://github.com/wso2/apim-analytics-publisher/issues/26

File: component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/auth/DefaultApi.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+import feign.RequestLine;
+
+/**
+ * Analytics Auth API
+ *
+ * <p>This is a RESTFul API for authenication information to communcate with event hub 
+ *
+ */
+public interface DefaultApi  {
```
Comment (jaadds) [inline component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/auth/DefaultApi.java:29]: Why do we call it DefaultAPI?

File: component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/client/EventHubProducerClientFactory.java (hunk 2 above)
Comment (ruks) [reply component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/client/EventHubProducerClientFactory.java]: Fixed

File: component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/auth/AuthClient.java (hunk 1 above)
Comment (ruks) [reply component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/auth/AuthClient.java]: Fixed

File: component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/auth/DefaultApi.java (hunk 4 above)
Comment (ruks) [reply component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/auth/DefaultApi.java:29]: I use a code generator and let use it. This is how it derived the name. But I can rename it.

**Review by tishan89 — APPROVED** [review]

**Pull request merged**